demo/c-api/README.md

```diff
@@ -0,0 +1,30 @@
+C-APIs
+===
+
+**XGBoost** implements a C API originally designed for various language
+bindings.  For detailed reference, please check xgboost/c_api.h.  Here is a
+demonstration of using the API.
+
+# CMake
```

Comment:
We need tests for this demo using cmake in a future PR. Functionality that is not tested is likely to be broken accidentally.

Reply:
@hcho3 is changing the ci I think?  To test this we need to create two dummy projects one for each case.

Reply:
@trivialfis Yes, I'm going to review this PR right now, and then file a PR to refactor Jenkins.